numpy/lib/recfunctions.py

```diff
@@ -523,6 +523,43 @@ def _drop_descr(ndtype, drop_names):
     return _fix_output(output, usemask=usemask, asrecarray=asrecarray)
 
 
+def _keep_fields(base, keep_names, usemask=True, asrecarray=False):
```

Comment:
Perhaps `dtype[['field1', 'field2']]` should be legal, to replace this function (but not needed for this PR, and overlaps with #8814).